spinner/spinner.go

```diff
@@ -61,7 +62,7 @@ func (s *Spinner) Title(title string) *Spinner {
 }
 
 // Action sets the action of the spinner.
-func (s *Spinner) Action(action func()) *Spinner {
+func (s *Spinner) Action(action func() error) *Spinner {
```

Comment:
The only issue you have with breaking changes is here.


```suggestion
func (s *Spinner) Action(action func()) *Spinner {
      return s.ActionErr(func() {
              action()
              return nil
    })
}

func (s *Spinner) ActionErr(action func()) *Spinner {
```